client/src/components/form-builder/form-field.tsx

```diff
@@ -0,0 +1,220 @@
+  useFormContext,
+  useWatch,
+} from 'react-hook-form';
+import {
+  Box,
+  Button,
+  Card,
+  CardContent,
+  Divider,
+  FormControl,
+  FormControlLabel,
+  Grid,
+  IconButton,
+  InputLabel,
+  MenuItem,
+  Select,
+  Switch,
+  TextField,
+} from '@mui/material';
+import {
+  Clear,
+  ContentCopy,
+  Delete,
+  RadioButtonUnchecked,
+} from '@mui/icons-material';
+import { secureRandomString } from '../../utils';
+import Spacer from '../spacer/Spacer';
+import { TextInput } from '../text-field/TextField';
+import { FieldTypeSelect } from './FieldTypeSelect';
+
+interface FormFieldProps {
+  sectionIndex: number;
+  index: number;
+  onRemove: () => void;
+}
+
+export const FormField = ({
+  index,
+  sectionIndex,
+  onRemove,
+}: FormFieldProps) => {
+  const n = (
+    name: keyof FormFieldSchema,
+  ): `sections.${number}.fields.${number}.${keyof FormFieldSchema}` =>
+    `sections.${sectionIndex}.fields.${index}.${name}`;
+  const { control, setValue, register } = useFormContext();
+  const [shouldShowDescription, setShouldShowDescription] = useState(false);
+  const type = useWatch({
+    control,
+    name: n('type'),
+  });
+  const options = useWatch({
+    control,
+    name: n('options'),
+  });
+  const isOptionQuestionType =
+    type === 'multipleChoice' || type === 'multiSelect';
+  const { append, remove, fields } = useFieldArray({
+    control,
+    name: `sections.${sectionIndex}.fields.${index}.options`,
+  });
+  return (
+    <Card sx={{ mb: 3 }} data-testid="form-field">
+      <CardContent>
+        <Grid container spacing={2}>
+          <Grid item xs={8}>
+            <TextInput
+              control={control}
+              name={n('name')}
+              label="Question title"
+              rules={{
+                required: true,
+                minLength: 1,
+              }}
+            />
+          </Grid>
+          <Grid item xs={4}>
+            <FieldTypeSelect
+              control={control}
+              name={n('type')}
+              onChange={(e) => {
+                if (
+                  e.target.value !== 'multipleChoice' &&
+                  e.target.value !== 'multiSelect'
+                ) {
+                  setValue(n('options'), []);
+                } else if (Array.isArray(options) && !options?.length) {
+                  setValue(n('options'), [
+                    {
+                      id: secureRandomString(12),
+                      label: '',
+                    } as any,
+                  ]);
+                }
+              }}
+            />
+          </Grid>
+          <Grid item xs={12}>
+            {shouldShowDescription && (
+              <TextInput
+                name={n('description')}
+                control={control}
+                label="Description (optional)"
+              />
+            )}
+          </Grid>
+          {isOptionQuestionType &&
+            Array.isArray(fields) &&
+            fields.map((f: any, i) => (
```

Comment:
same question with the `any` as above

edit: i see its because of the `isArray` function and the type guard (correct me if im wrong). There should be a way to abstract the `isArray` check out of here and only provide a boolean instead 

```
const checkArray = useMemo(() => Array.isArray(fields), [fields])
```

Reply:
So the type technically can be an object (dynamic data) or an array (static options), I will add the memoized version now.